tests/framework/data/ActiveDataProviderTest.php

```diff
@@ -177,4 +178,42 @@ public function testPaginationBeforeModels()
         $provider->refresh();
         $this->assertEquals(2, count($provider->getModels()));
     }
+
+    public function testZeroModels()
```

Comment:
lets keep it close to the code. I would prefer something like: `testSkipQueryWhenZeroTotalCount()`